climada/hazard/centroids/centr.py

```diff
@@ -79,280 +82,175 @@ class Centroids():
 
     Attributes
     ----------
-    meta : dict, optional
-        rasterio meta dictionary containing raster properties: width, height, crs and transform
-        must be present at least. The affine ransformation needs to be shearless (only stretching)
-        and have positive x- and negative y-orientation.
-    lat : np.array, optional
-        latitude of size size
-    lon : np.array, optional
-        longitude of size size
-    geometry : gpd.GeoSeries, optional
-        contains lat and lon crs. Might contain geometry points for lat and lon
+    lat : np.array
+        latitudes in the chosen crs (can be any unit)
+    lon : np.array
+        longitudes in the chosen crs (can be any unit)
+    crs : str, optional
+        coordinate reference system, default is WGS84
     area_pixel : np.array, optional
-        area of size size
+        areas
     dist_coast : np.array, optional
-        distance to coast of size size
+        distances to coast
     on_land : np.array, optional
-        on land (True) and on sea (False) of size size
+        on land (True) and on sea (False)
     region_id : np.array, optional
-        country region code of size size
+        region numeric codes
     elevation : np.array, optional
-        elevation of size size
+        elevations
+    kwargs: dicts of np.arrays, optional
+        any further desired properties of centroids. Is passed to the
+        GeoDataFrame constructor
     """
 
-    vars_check = {'lat', 'lon', 'geometry', 'area_pixel', 'dist_coast',
-                  'on_land', 'region_id', 'elevation'}
-    """Variables whose size will be checked"""
-
     def __init__(
         self,
-        lat: Optional[np.ndarray] = None,
-        lon: Optional[np.ndarray] = None,
-        geometry: Optional[gpd.GeoSeries] = None,
-        meta: Optional[Dict[Any, Any]] = None,
-        area_pixel: Optional[np.ndarray] = None,
-        on_land: Optional[np.ndarray] = None,
+        longitude: np.ndarray,
+        latitude: np.ndarray,
+        crs: str = DEF_CRS,
         region_id: Optional[np.ndarray] = None,
-        elevation: Optional[np.ndarray] = None,
-        dist_coast: Optional[np.ndarray] = None
+        on_land: Optional[np.ndarray] = None,
+        **kwargs
```

Comment:
I think the initializer should be the most basic form of defining the object. As such, I suggest to have an initializer that simply passes all its arguments directly to the `GeoDataFrame`. We can then define classmethods that initialize it from more "high-level" data, like so:
```python
class Centroids:

    def __init__(self, *args, **kwargs):
        self.gdf = GeoDataFrame(*args, **kwargs)

    @classmethod
    def from_lat_lon(cls, lat, lon, crs, *args, **kwargs):
        return cls(*args, geometry=gpd.points_from_xy(lon, lat, crs=crs), **kwargs)
```

Reply:
While I agree it should be basic, I really really do not like when the initializer is not clear. Also, this would require some major rewriting as I understand it. The current version specifies what is necessary (we want to force any user to ALWAYS define the crs for instance), plus two optional ones `region_id` and `on_land`. We could remove these two from the initializer if you prefer. It is however then not clear that these can be set at all, and that they are useful for many other applications. So, a user would have to know somehow that these attributes are to be called this way.

Reply:
Proposal after personal discussion: No arbitrary GeoDataFrame construction, but user can pass columns and *must* pass a geometry and crs.

Open questions:
* Is that feasible? Centroids are currently constructed from lat/lon coordinates. This changes would require refactoring of a lot of code. (`Centroids()` to `Centroids.from_lat_lon()`).
* Should we enforce the geometry type in the `__init__`? Not sure how costly this is for large frames. Maybe have an option to skip this if we are sure only points are passed?

```python
class Centroids:

    def __init__(self, geometry, crs, check_geometry=True, **kwargs):
        if check_geometry:
            if not (geometry.geom_type == shapely.Point).all():
                raise ValueError

        self.gdf = GeoDataFrame(geometry=geometry, crs=crs, data=kwargs)

    @classmethod
    def from_lat_lon(cls, lat, lon, crs, **kwargs):
        return cls(geometry=gpd.points_from_xy(lon, lat, crs=crs), check_geometry=False, **kwargs)
```